l10n_ch_delivery_carrier_label_quickpac/quickpac/web_service.py

```diff
@@ -0,0 +1,531 @@
+            label_layout=label_layout,
+        )
+        return label_definition
+
+    def _prepare_customer(self, picking):
+        """ Define the Quickpac direct client
+
+        :param picking: a picking record
+        :param company: The company sending the goods
+        :return: GenerateLabelCustomer
+        """
+        company = picking.company_id
+        if picking.picking_type_id.code == 'outgoing':
+            partner = company.partner_id
+        elif picking.picking_type_id.code == 'incoming':
+            partner = picking.partner_id
+
+        customer = GenerateLabelCustomer(
+            name1=partner.name,
+            street=partner.street,
+            zip=partner.zip,
+            city=partner.city,
+            country=partner.country_id.code,
+        )
+        if (
+            picking.picking_type_id.code == 'outgoing'
+            and company.quickpac_office
+        ):
+            customer.po_box = company.quickpac_office
+
+        if partner.parent_id and partner.parent_id.name != partner.name:
+            customer.name2 = customer.name1
+            customer.name1 = partner.parent_id.name
+
+        logo = get_logo(picking)
+        if logo:
+            bytes_logo = base64.b64decode(logo.decode())
+            logo_image = Image.open(io.BytesIO(bytes_logo))
+            logo_format = logo_image.format
+            customer.logo = logo.decode()
+            customer.logo_format = logo_format
+
+        return customer
+
+    def _prepare_file_infos(self, picking, company):
+        """ Define the sender informations
+
+        :param picking: a picking record
+        :param company: The company sending the goods
+        :return: GenerateLabelFileInfos
+        """
+        customer = self._prepare_customer(picking)
+        file_infos = GenerateLabelFileInfos(
+            mode="Label",
+            franking_license=company.quickpac_franking_license,
+            customer=customer,
+        )
+        return file_infos
+
+    def _prepare_recipient(self, picking):
+        """ Create a Recipient for a partner from a picking
+
+        :param picking: a picking record
+        :return: Recipient
+        """
+        partner = self._get_recipient_partner(picking)
+        partner_name = partner.name or partner.parent_id.name
+        email = parseaddr(partner.email)[1]
+        recipient = Recipient(
+            name1=sanitize_string(partner.name),
+            street=sanitize_string(partner.street),
+            zip=partner.zip,
+            city=partner.city,
+            e_mail=email or None,
+        )
+
+        if partner.country_id.code:
+            recipient.country = partner.country_id.code.upper()
+        if partner.street2:
+            recipient.addressSuffix = sanitize_string(partner.street2)
+
+        if partner.parent_id and partner.parent_id.name != partner_name:
+            recipient.name2 = sanitize_string(partner.parent_id.name)
+            recipient.personallyAddressed = False
```

Comment:
This should follow quickpac package definition `lib/python3.8/site-packages/quickpac/models/recipient.py:        'personally_addressed': 'PersonallyAddressed',`.
```suggestion
            recipient.personally_addressed = False
```